ndcube/ndcube_sequence.py

```diff
@@ -0,0 +1,170 @@
+
+    def __getitem__(self, item):
+        if item is None or (isinstance(item, tuple) and None in item):
+            raise IndexError("None indices not supported")
+        return cube_utils.get_cube_from_sequence(self, item)
+
+    def plot(self, *args, **kwargs):
+        i = ani.ImageAnimatorNDCubeSequence(self, *args, **kwargs)
+        return i
+
+    def to_sunpy(self, *args, **kwargs):
+        result = None
+        if all(isinstance(instance_sequence, sunpy.map.mapbase.GenericMap)
+               for instance_sequence in self.data):
+            result = MapCube(self.data, *args, **kwargs)
+        else:
+            raise NotImplementedError("Sequence type not Implemented")
+        return result
+
+    def explode_along_axis(self, axis):
+        """
+        Separates slices of NDCubes in sequence along a given cube axis into (N-1)DCubes.
+
+        Parameters
+        ----------
+
+        axis : `int`
+            The axis along which the data is to be changed.
+        """
+        # if axis is None then set axis as common axis.
+        if self._common_axis is not None:
+            if self._common_axis != axis:
+                raise ValueError("axis and common_axis should be equal.")
+        # is axis is -ve then calculate the axis from the length of the dimensions of one cube
+        if axis < 0:
+            axis = len(self.dimensions.shape[1::]) + axis
+        # To store the resultant cube
+        result_cubes = []
+        # All slices are initially initialised as slice(None, None, None)
+        result_cubes_slice = [slice(None, None, None)] * len(self[0].data.shape)
+        # the range of the axis that needs to be sliced
+        range_of_axis = self[0].data.shape[axis]
+        for ndcube in self.data:
+            for index in range(range_of_axis):
+                # setting the slice value to the index so that the slices are done correctly.
+                result_cubes_slice[axis] = index
+                # appending the sliced cubes in the result_cube list
+                result_cubes.append(ndcube.__getitem__(tuple(result_cubes_slice)))
+        # creating a new sequence with the result_cubes keeping the meta and common axis as axis
+        return self._new_instance(result_cubes, meta=self.meta, common_axis=axis)
+
+    def __repr__(self):
+        return (
+            """Sunpy NDCubeSequence
+---------------------
+Length of NDCubeSequence:  {length}
+Shape of 1st NDCube: {shapeNDCube}
+Axis Types of 1st NDCube: {axis_type}
+""".format(length=self.dimensions.shape[0], shapeNDCube=self.dimensions.shape[1::],
+                axis_type=self.dimensions.axis_types[1::]))
+
+    @property
+    def dimensions(self):
+        return SequenceDimensionPair(
+            shape=tuple([len(self.data)]+list(self.data[0].dimensions.shape)),
+            axis_types=tuple(["Sequence Axis"]+self.data[0].dimensions.axis_types))
+
+    @property
+    def _common_axis_extra_coords(self):
```

Comment:
For simplicity, could you make this property public as part of this PR?  Same for ```NDCube._extra_coords```.

Reply:
On second thoughts, I realised this may have impact on tests and will definitely have a knock onto IRISpy.  So I'll do it in a separate PR.